src/v0/destinations/klaviyo/transform.js

```diff
@@ -191,17 +187,16 @@ const trackRequestHandler = (message, category, destination) => {
     if (value) {
       attributes.value = value;
     }
-    attributes.properties = {
-      ...attributes.properties,
-      ...populateCustomFieldsFromTraits(message),
-    };
   }
   // if flattenProperties is enabled from UI, flatten the event properties
   attributes.properties = flattenProperties
     ? flattenJson(attributes.properties, '.', 'normal', false)
     : attributes.properties;
   // Map user properties to profile object
-  attributes.profile = createCustomerProperties(message, destination.Config);
+  attributes.profile = {
+    ...createCustomerProperties(message, destination.Config),
+    ...flattenJson(populateCustomFieldsFromTraits(message)),
```

Comment:
Having profile properties in flattened json by default is preferable ?

Reply:
This is not the case in identify flow

Reply:
Yes, this is preferable and we are doing it in `identify` as well in the line you pointed out in the next comment.

Reply:
But it's driven via `flattenProperties` flag not by default

Reply:
Yeah, Misunderstood it. Updated it via `flattenProperties`.